hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/client/SparkRDDWriteClient.java

```diff
@@ -100,7 +107,9 @@ public static SparkConf registerClasses(SparkConf conf) {
   public boolean commit(String instantTime, JavaRDD<WriteStatus> writeStatuses, Option<Map<String, String>> extraMetadata,
                         String commitActionType, Map<String, List<String>> partitionToReplacedFileIds) {
     List<HoodieWriteStat> writeStats = writeStatuses.map(WriteStatus::getStat).collect();
-    return commitStats(instantTime, writeStats, extraMetadata, commitActionType, partitionToReplacedFileIds);
+    boolean isCommitted = commitStats(instantTime, writeStats, extraMetadata, commitActionType, partitionToReplacedFileIds);
+    this.hoodieHeartbeatClient.stop(instantTime);
```

Comment:
hmmm. we should probably do this only once at the super class level.

Reply:
Moved to `postCommit`